teuthology/task/install/rpm.py

```diff
@@ -179,28 +175,21 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
                 cpack=cpack,
             )
             remote.run(
-                args=['if', 'test', '-e',
-                      run.Raw(pkg), run.Raw(';'), 'then',
-                      'sudo', pkg_mng_cmd, pkg_mng_opts, 'remove',
-                      pkg_mng_subcommand_opts, pkg, run.Raw(';'),
-                      'sudo', pkg_mng_cmd, pkg_mng_opts, pkg_mng_gpg_opt, 'install',
-                      pkg_mng_subcommand_opts, pkg_mng_install_opts,
-                      pkg, run.Raw(';'),
-                      'fi']
+                args=chain(
+                    ['if', 'test', '-e', run.Raw(pkg), run.Raw(';'), 'then'],
+                    pkg_remove_cmd.format(pkg=pkg).split(), [run.Raw(';')],
+                    pkg_install_cmd.format(pkg=pkg).split(), [run.Raw(';')],
+                    ['fi'])
             )
         if pkg is None:
```

Comment:
Reverse logic, and same as above.
```
if pkg:
    if dist_release in ['opensuse', 'sle']:
        remote.run(args="""
            if test ! -e {package} ; then
                sudo zypper -n --no-gpg-checks install --capability --no-recommends {package}
            fi
        """.format(package=pkg)
    else:
        remote.run(args="""
            if test ! -e {package} ; then
                sudo yum -y install {package}
            fi
        """.format(package=pkg)
else:
    if dist_release in ['opensuse', 'sle']:
        remote.run(args="""
            sudo zypper -n --no-gpg-checks install --capability --no-recommends {package}
        """.format(package=cpack)
    else:
        remote.run(args="sudo yum -y install %s" % cpack)
```

Reply:
@kshtsk it's out of scope of this change. as i think the focus of it is not really refactoring, it's more about fixing the rpm install issue on centos. we can always fix the reverse logic in a follow-up PR.

Reply:
I agree with @tchaikov - this fix LGTM

Reply:
I agree too